src/scripts/main.js

```diff
@@ -1,11 +1,25 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve, reject) => {
+    element.addEventListener(
+      eventName,
+      (e) => {
+        resolve(
+          `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`,
+        );
```

Comment:
The information resolved in the Promise is very detailed, which is good for debugging, but consider if all this information is needed every time the event is triggered. If not, you might want to resolve with less information to avoid potential performance issues with unnecessary string concatenations.